packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,34 +458,89 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    private async fetchSnapshot(snapshotOptions: string, tokenFetchOptions: TokenFetchOptions) {
-        const storageToken = await this.getStorageToken(tokenFetchOptions, "TreesLatest");
+    private async fetchSnapshot(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions) {
+        const usePost = this.hostPolicy.usePostForTreesLatest;
+        // If usePost is false, then make get call for TreesLatest.
+        // If usePost is true, and token refresh is false, then it means we are trying it first time, so
+        // don't catch the error as getWithRetryForTokenRefresh will handle and try with refresh.
+        // If usePost is true and we are trying with a refreshed token, then use the fallback code in catch.
+        if (usePost && tokenFetchOptions.refresh) {
```

Comment:
I'm not sure I agree with logic. Refresh happens if token has expired. If it's not then there is no second chance.
Given that we do not have fallback post to get on first attempt, I'm not sure why it makes sense to have fallback on second attempt. I.e. it seems like they need to be in sync - we either always do fallback, or we never do fallback.

Reply:
So the logic now is we make a post call and if that fails for a reason for which getWithRetryForTokenRefresh will not retry we fallback to get call. otherwise we let getWithRetryForTokenRefresh to refresh token for post call and then we make post call again. Now second time if the post call fail again we fallback to get call.